src/BasketBundle/Form/PaymentType.php

```diff
@@ -99,7 +100,11 @@ public function buildForm(FormBuilderInterface $builder, array $options)
 
         $sub = $builder->create('paymentMethod', $choiceType, array(
             'expanded' => true,
-            'choice_list' => new SimpleChoiceList($choices),
+            'choice_list' => function (Options $opts, $previousValue) use ($choices) {
```

Comment:
There is something that I don't understand;
If I adapt you code to my situation like this:
```
if (!interface_exists('Symfony\Component\Form\ChoiceList\Loader\ChoiceLoaderInterface')) {
            $sub = $builder->create('deliveryMethod', $choiceType, array(
                'expanded' => true,
                'choice_list' => new SimpleChoiceList($choices),
            ));
        }else {
            $sub = $builder->create('deliveryMethod', $choiceType, array(
                'expanded' => true,
                'choice_loader' => function (Options $opts, $previousValue) use ($choices) {
                    return new DeliveryChoiceLoader(array_flip($choices));
                },
            ));
        }
```
I get the following error:
`The option "choice_loader" with value Closure is expected to be of type "null" or "Symfony\Component\Form\ChoiceList\Loader\ChoiceLoaderInterface", but is of type "Closure".`

instead, if I change it like this:
```
if (!interface_exists('Symfony\Component\Form\ChoiceList\Loader\ChoiceLoaderInterface')) {
            $sub = $builder->create('deliveryMethod', $choiceType, array(
                'expanded' => true,
                'choice_list' => new SimpleChoiceList($choices),
            ));
        }else {
            $sub = $builder->create('deliveryMethod', $choiceType, array(
                'expanded' => true,
                'choice_loader' => new DeliveryChoiceLoader(array_flip($choices))
            ));
        }
```

it seems to be working